### DISCUSSION TITLE
Default language setting?

### DISCUSSION THREAD

**Discussion by psharp15** [zed-industries/zed#6536] [Feature Requests]
Is there a way to set a default language setting? I want Zed to assume I'm writing in C++ unless otherwise changed.

**Comment by hovsater** [comment]
This isn't possible right now. Zed will automatically detect what language to use based on the current file extension. You can also manually set the language by invoking `⌘+K M`.

**Reply by sullimander** [reply]
@hovsater is there a way to set the language for file extensions manually? I would like to change `*.jbuilder` files to always be Ruby.

**Reply by vurihuang** [reply]
#10765 already released the feature of regex filetype, you can try these, it works for me: 

```json5
{
  "file_types": {
    "Shell Script": [
      ".aliases"
    ],
    "C++": [
      "*.jbuilder"
    ]
  }
}
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
